src/neuroglancer/ui/url_hash_binding.ts

```diff
@@ -98,7 +98,7 @@ export class UrlHashBinding extends RefCounted {
         s = '#!{}';
       }
       // Handle remote JSON state
-      if (s.match(/^#!(http|https|gs):\/\//)) {
+      if (s.match(/^#!(http|https|gs|middleauth\+https):\/\//)) {
```

Comment:
Probably reasonable to just match anything that looks like a protocol -- that way we don't need the list of special protocols duplicated here.  I think I also forgot to properly support ngauth here as well...